tools/runk/src/state.rs

```diff
@@ -0,0 +1,104 @@
+// SPDX-License-Identifier: Apache-2.0
+//
+// Copyright 2021 Sony Group Corporation
+//
+
+use crate::status::{self, Status};
+use anyhow::Result;
+use chrono::{DateTime, Utc};
+use nix::unistd::Pid;
+use oci::ContainerState;
+use rustjail::container::EXEC_FIFO_FILENAME;
+use serde::{Deserialize, Serialize};
+use slog::{info, Logger};
+use std::path::Path;
+
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct State {
+    pub oci_version: String,
+    pub id: String,
+    pub pid: i32,
+    pub status: String,
+    pub bundle: String,
+    pub created: DateTime<Utc>,
+}
+
+impl State {
+    pub fn new(status: Status, state: ContainerState) -> Self {
+        Self {
+            oci_version: status.oci_version,
+            id: status.id,
+            pid: status.pid,
+            status: get_state_name_string(state),
+            bundle: status.bundle,
+            created: status.created,
+        }
+    }
+}
+
+pub fn state_command(id: &str, state_root: &Path, logger: &Logger) -> Result<()> {
+    let status = Status::load(state_root, id)?;
+    let state = get_current_container_state(&status)?;
+    let oci_state = State::new(status, state);
+    let json_str = &serde_json::to_string_pretty(&oci_state)?;
+
+    println!("{}", json_str);
+
+    info!(&logger, "state command finished successfully");
+
+    Ok(())
+}
+
+fn get_state_name_string(state: ContainerState) -> String {
+    match state {
+        ContainerState::Creating => "creating",
+        ContainerState::Created => "created",
+        ContainerState::Running => "running",
+        ContainerState::Stopped => "stopped",
+        ContainerState::Paused => "paused",
+    }
+    .into()
+}
+
+pub fn get_current_container_state(status: &Status) -> Result<ContainerState> {
+    let running = status::is_process_running(Pid::from_raw(status.pid))?;
+    let mut has_fifo = false;
+
+    if running {
+        let fifo = format!("{}/{}/{}", status.bundle, status.id, EXEC_FIFO_FILENAME);
```

Comment:
What if `status.id == ""`?

Reply:
When launching container using `runk`, it is mandatory to input the `container id` (empty string is prohibited) in the CLI, so I think that `status.id` won't be empty.